src/rosdistro/rosdistro.py

```diff
@@ -204,20 +204,21 @@ def _fetch_package_xml(self, rosdistro):
             url = url.replace('.git', '/{0}/package.xml'.format(release_tag))
             url = url.replace('git://', 'https://')
             url = url.replace('https://', 'https://raw.')
+            legacy_url = url # save the old type of url, try it second.
+
             try:
                 try:
-                    package_xml = urlopen(url).read()
-                except Exception as e:
-                    msg = "Failed to read package.xml file from url '{0}': {1}".format(url, e)
-                    warning(msg)
                     url = repo.url
                     release_tag = 'release/{0}/{1}/{2}'.format(rosdistro, self.name, repo.version)
                     tail = '/{0}/package.xml'.format(release_tag)
                     url = url.replace('.git', tail)
-                    url = url.replace('git://', 'https://')
```

Comment:
It was not intentional. Good catch. Oddly, it works, but... I haven't tried it for indigo yet. One moment.